tensorboard/webapp/metrics/views/right_pane/scalar_column_editor/scalar_column_editor_component.ts

```diff
@@ -25,27 +26,126 @@ import {
   DataTableMode,
 } from '../../card_renderer/scalar_card_types';
 
+const preventDefault = (e: MouseEvent) => {
+  e.preventDefault();
+};
+
+// Move the item at sourceIndex to destinationIndex
+const moveHeader = (
+  sourceIndex: number,
+  destinationIndex: number,
+  headers: ColumnHeader[]
+) => {
+  const newHeaders = [...headers];
+  // Delete from original location
+  newHeaders.splice(sourceIndex, 1);
+  // Insert at destinationIndex.
+  newHeaders.splice(destinationIndex, 0, headers[sourceIndex]);
+  return newHeaders;
+};
+
+const getIndexOfType = (type: ColumnHeaderType, headers: ColumnHeader[]) => {
+  return headers.findIndex((header) => {
+    return header.type === type;
+  });
+};
+
+enum Edge {
+  TOP,
+  BOTTOM,
+}
+
 @Component({
   selector: 'metrics-scalar-column-editor-component',
   templateUrl: 'scalar_column_editor_component.ng.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
   styleUrls: [`scalar_column_editor_component.css`],
 })
-export class ScalarColumnEditorComponent {
+export class ScalarColumnEditorComponent implements OnDestroy {
   DataTableMode = DataTableMode;
   selectedTab: DataTableMode = DataTableMode.SINGLE;
+  draggingHeaderType: ColumnHeaderType | undefined;
+  highlightedHeaderType: ColumnHeaderType | undefined;
+  highlightEdge: Edge = Edge.TOP;
   @Input() rangeHeaders!: ColumnHeader[];
   @Input() singleHeaders!: ColumnHeader[];
 
+  @Output() onScalarTableColumnEdit = new EventEmitter<{
+    dataTableMode: DataTableMode;
+    headers: ColumnHeader[];
+  }>();
   @Output() onScalarTableColumnToggled = new EventEmitter<{
     dataTableMode: DataTableMode;
     headerType: ColumnHeaderType;
   }>();
 
+  ngOnDestroy() {
+    document.removeEventListener('dragover', preventDefault);
+  }
```

Comment:
Is there a reason you need to add this to the entire document? I imagine preventing the default within this container should be enough.

```suggestion
  constructor(private readonly hostElement: ElementRef) {}
 
  ngOnDestroy() {
    this.hostElement.removeEventListener('dragover', preventDefault);
  }
```